modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/display/context/CPInstanceUnitOfMeasureDisplayContext.java

```diff
@@ -0,0 +1,185 @@
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.portlet.LiferayWindowState;
+import com.liferay.portal.kernel.portlet.url.builder.PortletURLBuilder;
+import com.liferay.portal.kernel.util.Constants;
+import com.liferay.portal.kernel.util.ListUtil;
+
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * @author Crescenzo Rega
+ */
+public class CPInstanceUnitOfMeasureDisplayContext
+	extends BaseCPDefinitionsDisplayContext {
+
+	public CPInstanceUnitOfMeasureDisplayContext(
+		ActionHelper actionHelper, HttpServletRequest httpServletRequest,
+		CPInstanceUnitOfMeasureService cpInstanceUnitOfMeasureService) {
+
+		super(actionHelper, httpServletRequest);
+
+		_cpInstanceUnitOfMeasureService = cpInstanceUnitOfMeasureService;
+	}
+
+	public List<DropdownItem> getBulkActionDropdownItems() {
+		return ListUtil.fromArray(
+			new FDSActionDropdownItem(
+				PortletURLBuilder.createActionURL(
+					cpRequestHelper.getRenderResponse()
+				).setActionName(
+					"/cp_definitions/edit_cp_instance_unit_of_measure"
+				).setCMD(
+					Constants.DELETE
+				).setRedirect(
+					cpRequestHelper.getCurrentURL()
+				).buildString(),
+				"trash", "delete", "delete",
+				LanguageUtil.get(httpServletRequest, "delete"), "delete",
+				null));
+	}
+
+	public String getCommerceCurrencyCode() throws PortalException {
+		CPInstance cpInstance = getCPInstance();
+
+		CommerceCatalog commerceCatalog = cpInstance.getCommerceCatalog();
+
+		return commerceCatalog.getCommerceCurrencyCode();
+	}
+
+	public CPInstance getCPInstance() throws PortalException {
+		if (_cpInstance != null) {
+			return _cpInstance;
+		}
+
+		_cpInstance = actionHelper.getCPInstance(
+			cpRequestHelper.getRenderRequest());
+
+		return _cpInstance;
+	}
+
+	public long getCPInstanceId() throws PortalException {
+		CPInstance cpInstance = getCPInstance();
+
+		if (cpInstance == null) {
+			return 0;
+		}
+
+		return cpInstance.getCPInstanceId();
+	}
+
+	public CPInstanceUnitOfMeasure getCPInstanceUnitOfMeasure()
+		throws PortalException {
+
+		if (_cpInstanceUnitOfMeasure != null) {
+			return _cpInstanceUnitOfMeasure;
+		}
+
+		_cpInstanceUnitOfMeasure = actionHelper.getCPInstanceUnitOfMeasure(
+			cpRequestHelper.getRenderRequest());
+
+		return _cpInstanceUnitOfMeasure;
+	}
+
+	public CreationMenu getCreationMenu() throws Exception {
+		return CreationMenuBuilder.addDropdownItem(
+			dropdownItem -> {
+				dropdownItem.setHref(
+					PortletURLBuilder.createRenderURL(
+						liferayPortletResponse
+					).setMVCRenderCommandName(
+						"/cp_definitions/add_cp_instance_unit_of_measure"
+					).setBackURL(
+						cpRequestHelper.getCurrentURL()
+					).setParameter(
+						"cpInstanceId", getCPInstanceId()
+					).setWindowState(
+						LiferayWindowState.POP_UP
+					).buildString());
+				dropdownItem.setLabel(
+					LanguageUtil.get(
+						cpRequestHelper.getRequest(), "add-value"));
```

Comment:
You can use `httpServletRequest` instead of getting it from `cpRequestHelper` like [here](https://github.com/liferay-commerce/liferay-portal/pull/4208/commits/403759eb6ce89efa42f1da427212da2a9d6ef26b#diff-d1818d7860fd9463339846315985b8c6ac95d4439e0e59dfe97a20cec6318a80R58)